robo_soccer_platform/lib/robo_soccer_platform/game_controller.ex

```diff
@@ -0,0 +1,315 @@
+          |> Map.values()
+          |> Enum.filter(&(&1.player.team == team))
+          |> aggregate_player_inputs(
+            state.aggregation_function,
+            state.speed_coefficient
+          )
+
+        RobotConnection.send_instruction(robot_connection, instruction)
+        {team, instruction}
+      end)
+
+    RoboSoccerPlatformWeb.GameDashboard.update_steering_state(state.game_dashboard_pid, %{
+      player_inputs: state.player_inputs,
+      robot_instructions: robot_instructions
+    })
+
+    {:noreply,
+     %{
+       state
+       | robot_instructions: robot_instructions,
+         aggregation_timer: start_aggregation_timer(state.aggregation_interval_ms)
+     }}
+  end
+
+  @impl true
+  def handle_info(msg, state) do
+    Logger.warning("""
+    [#{inspect(__MODULE__)}] Ignoring message: #{inspect(msg)}
+    State: #{inspect(state)}
+    """)
+
+    {:noreply, state}
+  end
+
+  @spec init_state(options()) :: State.t()
+  defp init_state(opts) do
+    aggregation_interval_ms =
+      Keyword.get(opts, :aggregation_interval_ms, @default_aggregation_interval_ms)
+
+    aggregation_function_name =
+      Keyword.get(opts, :aggregation_function_name, @default_aggregation_function_name)
+
+    speed_coefficient =
+      Keyword.get(opts, :speed_coefficient, @default_speed_coefficient)
+
+    robot_connections =
+      opts
+      |> Keyword.fetch!(:robot_configs)
+      |> Map.new(fn {team, %{robot_ip_address: ip_address, local_port: local_port}} ->
+        {:ok, ip_address} = ip_address |> String.to_charlist() |> :inet.parse_address()
+        {local_port, ""} = Integer.parse(local_port)
+
+        case RobotConnection.start(team, local_port, ip_address) do
+          {:ok, pid} ->
+            Logger.info("Started RobotConnection for team #{team} with pid #{inspect(pid)}")
+            {team, pid}
+
+          {:error, reason} ->
+            raise "Failed starting RobotConnection for team #{team}, reason: #{reason}"
+        end
+      end)
+
+    robot_instructions =
+      Map.new(robot_connections, fn {team, _robot_connection} -> {team, %{x: 0.0, y: 0.0}} end)
+
+    %State{
+      robot_instructions: robot_instructions,
+      robot_connections: robot_connections,
+      aggregation_interval_ms: aggregation_interval_ms,
+      speed_coefficient: speed_coefficient,
+      room_code: Enum.random(1000..9999) |> to_string,
+      aggregation_function:
+        Function.capture(RoboSoccerPlatform.AggregationFunctions, aggregation_function_name, 1)
+    }
+  end
+
+  @spec aggregate_player_inputs([player_input()], ([integer()] -> float()), float()) ::
+          %{x: float(), y: float()}
+  defp aggregate_player_inputs(player_inputs, aggregation_function, speed_coefficient) do
+    %{
+      x:
+        Enum.map(player_inputs, & &1.x)
+        |> then(aggregation_function),
+      y:
+        Enum.map(player_inputs, & &1.y)
+        |> then(aggregation_function)
+        |> Kernel.*(speed_coefficient)
```

Comment:
Shouldn't speed coefficient be multiplied by both X and Y? If we multiply only Y doesn't it change direction?

Reply:
This might change when we start supporting different robots, when that happens we should consider having different parameters for different models.